tool/gravity/cli/clusterupdate.go

```diff
@@ -183,8 +185,86 @@ func executeUpdatePhaseForOperation(env *localenv.LocalEnvironment, environ Loca
 		return trace.Wrap(err)
 	}
 	defer updater.Close()
-	err = updater.RunPhase(context.TODO(), params.PhaseID, params.Timeout, params.Force)
-	return trace.Wrap(err)
+	return executeOrForkPhase(env, updater, params, operation)
+}
+
+// gravityResumeServiceName is the name of systemd service that executes
+// the gravity resume command.
+const gravityResumeServiceName = "gravity-resume.service"
+
+// executeOrForkPhase either directly executes the specified operation phase,
+// or launches a one-shot systemd service that executes it in the background.
+func executeOrForkPhase(env *localenv.LocalEnvironment, updater updater, params PhaseParams, operation ops.SiteOperation) error {
+	// If given the --block flag, we're running as a systemd unit (or a user
+	// requested the command to execute in foreground), so proceed to perform
+	// the command (resume or single phase) directly.
+	if params.Block {
+		return updater.RunPhase(context.TODO(),
+			params.PhaseID,
+			params.Timeout,
+			params.Force)
+	}
+	// Before launching the service, perform a few prechecks, for example to
+	// make sure that the operation is being resumed from the correct node.
+	//
+	// TODO(r0mant): Also, make sure agents are running on the cluster nodes:
+	// https://github.com/gravitational/gravity/issues/1667
+	if err := updater.Check(params.toFSM()); err != nil {
+		return trace.Wrap(err)
+	}
+	// Make sure to launch the unit command with the --block flag.
+	args := append(os.Args[1:], "--debug", "--block")
```

Comment:
I think this is susceptible to the same issue as fixed [here](https://github.com/gravitational/gravity/pull/1255) - something to watch for and optionally rewrite with changes from https://github.com/gravitational/gravity/pull/1895.

Reply:
I don't think it's susceptible to the same issue. In the cases you're referring to we had to either replace the flags (remove/add), or update their values. In this case we just need to append a `--block` flag. Worst case scenario, it'll end up with two `--debug` flags which is fine. So unless I'm missing anything, I'd like to keep this simple.